trie/sync.go

```diff
@@ -352,6 +366,12 @@ func (s *Sync) Commit(dbw ethdb.Batch) error {
 		owner, inner := ResolvePath([]byte(path))
 		rawdb.WriteTrieNode(dbw, owner, inner, s.membatch.hashes[path], value, s.scheme)
 	}
+	for path := range s.membatch.deletes {
+		owner, inner := ResolvePath([]byte(path))
+		rawdb.DeleteTrieNode(dbw, owner, inner, common.Hash{} /* unused */, s.scheme)
+	}
```

Comment:
Shouldn't you do the deletions before the Writes? In case we're about to overwrite some parts of the path, the deletions might contain _all_ parts, and we don't want to delete the things we just wrote. Right?

Reply:
I was also thinking about it. IMO the deletions should only contain the short-key-internal parts, so it should not duplicate any existing new node we've just written.

Reply:
Yes, for each written node and deleted node, the path should be unique.